subsys/bluetooth/controller/hci_cmd_put.c

```diff
@@ -0,0 +1,851 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bluetooth/hci.h>
+#include <bluetooth/hci_err.h>
+#include <sdc_hci.h>
+#include <sdc_hci_cmd_controller_baseband.h>
+#include <sdc_hci_cmd_info_params.h>
+#include <sdc_hci_cmd_le.h>
+#include <sdc_hci_cmd_link_control.h>
+#include <sdc_hci_cmd_status_params.h>
+#include <sdc_hci_vs.h>
+
+#include "hci_cmd_put.h"
+
+#define CMD_COMPLETE_MIN_SIZE (BT_HCI_EVT_HDR_SIZE \
+				+ sizeof(struct bt_hci_evt_cmd_complete) \
+				+ sizeof(uint8_t))
+
+/* A host is not allowed to use both legacy and extended HCI commands.
+ * See Core v5.1, Vol2, Part E, 3.1.1 Legacy and extended advertising
+ */
+static enum {
+	ADV_COMMAND_TYPE_NONE,
+	ADV_COMMAND_TYPE_LEGACY,
+	ADV_COMMAND_TYPE_EXTENDED,
+} type_of_adv_cmd_used_since_reset;
+
+static struct
+{
+	bool occurred; /**< Set in only one execution context */
+	uint8_t raw_event[HCI_EVENT_PACKET_MAX_SIZE];
+} cmd_complete_or_status;
+
+static bool command_generates_command_complete_event(uint16_t hci_opcode)
+{
+	switch (hci_opcode) {
+	case SDC_HCI_OPCODE_CMD_LC_DISCONNECT:
+	case SDC_HCI_OPCODE_CMD_LE_SET_PHY:
+	case SDC_HCI_OPCODE_CMD_LC_READ_REMOTE_VERSION_INFORMATION:
+	case SDC_HCI_OPCODE_CMD_LE_CREATE_CONN:
+	case SDC_HCI_OPCODE_CMD_LE_CONN_UPDATE:
+	case SDC_HCI_OPCODE_CMD_LE_READ_REMOTE_FEATURES:
+	case SDC_HCI_OPCODE_CMD_LE_ENABLE_ENCRYPTION:
+	case SDC_HCI_OPCODE_CMD_LE_EXT_CREATE_CONN:
+		return false;
+	default:
+		return true;
+	}
+}
+
+/* Return true if the host has been using both legacy and extended HCI commands
```

Comment:
How do you decide whether to document a static function or not?

Reply:
For the cases that function is not trivial, I have added some docs